prime-router/src/main/kotlin/credentials/CredentialService.kt

```diff
@@ -1,34 +1,71 @@
 package gov.cdc.prime.router.credentials
 
+import io.ktor.client.HttpClient
 import org.apache.logging.log4j.kotlin.Logging
 
-abstract class CredentialService : Logging {
+abstract class CredentialService() : Logging {
 
     companion object {
         private val URL_SAFE_KEY_PATTERN = Regex("^[a-zA-Z0-9-]*$")
     }
 
     /* Methods to implement in subclasses */
 
-    protected abstract fun fetchCredential(connectionId: String): Credential?
-    protected abstract fun saveCredential(connectionId: String, credential: Credential)
+    abstract fun fetchCredential(
+        connectionId: String,
+        httpClient: HttpClient? = null,
```

Comment:
I'm not sure these belong on the interface?  It looks like only the `HashicorpVaultCredential` implementation would use them.